integration-tests/smoke/chainaccessor/accessor_test.go

```diff
@@ -0,0 +1,311 @@
+
+	updateFeeTokensMsg := feequoter.UpdateFeeTokens{
+		Add:    feeTokenDict,
+		Remove: tonCommon.SnakeData[*address.Address]{}, // Empty remove list
+	}
+
+	updateFeeTokensCell, err := tlb.ToCell(updateFeeTokensMsg)
+	require.NoError(t, err, "failed to encode UpdateFeeTokens message")
+
+	updateFeeTokensInternalMsg := &wallet.Message{
+		Mode: 1,
+		InternalMessage: &tlb.InternalMessage{
+			IHRDisabled: true,
+			Bounce:      false,
+			DstAddr:     &feeQuoterAddr,
+			Amount:      tlb.MustFromTON("0.01"),
+			Body:        updateFeeTokensCell,
+		},
+	}
+
+	tt := tracetracking.NewSignedAPIClient(tonChain.Client, *deployer)
+	updateFeeTokensResult, updateFeeTokensBlockID, err := tt.SendWaitTransaction(ctx, feeQuoterAddr, updateFeeTokensInternalMsg)
+	require.NoError(t, err, "failed to send UpdateFeeTokens transaction")
+
+	t.Logf("UpdateFeeTokens transaction sent successfully - Block: %d, ExitCode: %d",
+		updateFeeTokensBlockID.SeqNo, updateFeeTokensResult.ExitCode)
+
+	// TODO: missing in changeset ------------------ end
+
+	// TODO: use sendmanytx or highload wallet, otherwise we get 33 exit code(too many actions)
+	time.Sleep(5 * time.Second)
+
+	const lastSeqNo = 4
+	for seqNo := 0; seqNo <= lastSeqNo; seqNo++ {
+		t.Log("Sending CCIP message", seqNo)
+		sendCCIPMessage(t, evmSelector, routerAddr, tonChain.Client, deployer)
+		time.Sleep(2 * time.Second)
+	}
+
+	t.Run("query CCIP events via TonAccessor", func(t *testing.T) {
+		// check the latest message is indexed
+		require.Eventually(t, func() bool {
+			seqNum, err := accessor.LatestMessageTo(ctx, ccipocr3.ChainSelector(evmSelector))
+			require.NoError(t, err, "failed to get latest message sequence number")
+			return seqNum == ccipocr3.SeqNum(lastSeqNo)
+		}, 30*time.Second, 3*time.Second, "log poller did not ingest events correctly in time")
+
+		// check all messages are indexed
+		msgs, err := accessor.MsgsBetweenSeqNums(ctx, ccipocr3.ChainSelector(evmSelector), ccipocr3.NewSeqNumRange(0, lastSeqNo))
+		require.NoError(t, err, "failed to get latest message sequence number")
+		require.Len(t, msgs, lastSeqNo+1, "expected %d messages, got %d", lastSeqNo+1, len(msgs))
+		require.Equal(t, msgs[0].Header.SequenceNumber, ccipocr3.SeqNum(0))
+		require.Equal(t, msgs[lastSeqNo].Header.SequenceNumber, ccipocr3.SeqNum(lastSeqNo))
+
+		// range query
+		const start, end = 2, 4
+		msgs2, err := accessor.MsgsBetweenSeqNums(ctx, ccipocr3.ChainSelector(evmSelector), ccipocr3.NewSeqNumRange(start, end))
+		require.NoError(t, err, "failed to get latest message sequence number")
+		require.Len(t, msgs2, end-start+1, "expected %d messages, got %d", end-start+1, len(msgs2))
+		require.Equal(t, msgs2[0].Header.SequenceNumber, ccipocr3.SeqNum(start))
+		require.Equal(t, msgs2[len(msgs2)-1].Header.SequenceNumber, ccipocr3.SeqNum(end))
+	})
+}
+
+// TODO: clean up
+func sendCCIPMessage(t *testing.T, evmSelector uint64, routerAddr address.Address, client ton.APIClientWrapped, deployer *wallet.Wallet) {
```

Comment:
Let's use `SendRequest`, @huangzhen1997 is working on it here: https://github.com/smartcontractkit/chainlink-ton/pull/123